plaidml2/bridge/keras/__init__.py

```diff
@@ -1060,16 +1060,24 @@ def not_equal(lhs, rhs):
 
 @_log_call
 def normalize_batch_in_training(x, gamma, beta, reduction_axes, epsilon=1e-3):
-    I = x.tensor
-    ndims = I.shape.ndims
-    if reduction_axes == None:
-        raw_axes = [ndims - 1]
+    ndims = ndims(x)
+    if ndims == 4 and reduction_axes in [[0, 1, 2], [0, 2, 3]]:
+        target_shape = [1 if i in reduction_axes else shape(x) for i in range(4)]
```

Comment:
This should probably be `shape(x)[i]`. Also, I think this calls `shape` for each iteration of the list comprehension loop. I think it would be better to precompute shape(x) before the list comprehension and then access different elements of the precomputed value during the list comprehension.